packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -587,34 +672,248 @@ const changeCheckedGroup = () => {
       form.fields.port = portValue || getPort.getPortFromProtocol(form.fields.protocol)
 
-      form.errorMessage = ''
       emit('url-valid:success')
     } catch (error: any) {
-      form.errorMessage = t('errors.urlErrorMessage')
-      emit('url-valid:error', getMessageFromError(error))
+      emit('url-valid:error', error.message || 'URL validation failed')
+      form.formFieldErrors.url = error.message || 'URL validation failed'
     }
-  } else {
+  }
+}, 300)
+
+const handleValidateAdvancedFields = useDebounceFn((fieldId?: keyof FormFieldErrors) => {
+  // reset the errors
+  resetFormFieldErrors(fieldId ?? undefined)
+
+}, 300)
+
+const getFieldErrorById = computed(() => {
+  return (fieldId: keyof FormFieldErrors): string => {
+    const errorsMap = form.formFieldErrors
+
+    if (fieldId === 'url') {
+      if (errorsMap.host.length) return errorsMap.host
+      else if (errorsMap.path.length) return errorsMap.path
+      else return errorsMap.url
+    }
+
+    if (errorsMap[fieldId]) return errorsMap[fieldId]
+    return ''
+  }
+})
+
+// validate for the service type custom URL
+const handleValidateCustomUrl = useDebounceFn((fieldId?: keyof FormFieldErrors): void => {
+
+  // reset the errors
+  resetFormFieldErrors(fieldId ?? undefined)
+
+  // validate hostname
+  const hostError = validateHost(form.fields.host)
+  if (hostError) form.formFieldErrors.host = hostError
+
+  // validate path
+  const pathError = validatePath(form.fields.path)
+  if (pathError) form.formFieldErrors.path = pathError
+
+  // validate port
+  const portError = validatePort(form.fields.port)
+  if (portError) form.formFieldErrors.port = portError
+
+  if (isFormValid.value) {
     emit('url-valid:success')
-    form.errorMessage = ''
+  } else {
+    emit('url-valid:error', form.errorMessages.join(',') || 'URL validation failed')
+  }
+
+}, 300)
+
+const resetFormFieldErrors = (fieldId?: keyof FormFieldErrors): void => {
+  // if field Id is present only reset the field error
+  if (fieldId) {
+    form.formFieldErrors[fieldId] = ''
+  } else {
+    // reset all field errors
+    form.formFieldErrors = {
+      host: '',
+      port: '',
+      url: '',
+      path: '',
+      tags: '',
+      retries: '',
+      connect_timeout: '',
+      write_timeout: '',
+      read_timeout: '',
+      client_certificate: '',
+      ca_certificates: '',
+      tls_verify_enabled: '',
+      tls_verify_value: '',
+    }
   }
+  // reset the form Error TODO: clear the errors based on the field sent
+  form.errorMessages = []
+}
+
+const isFormValid = computed((): boolean => {
+  // check if no errors present
+  return Object.values(form.formFieldErrors).some(error => error !== null && error !== '') && !hasPreValidateError.value
+})
+
+const validateUrl = (): void => {
+  if (form.fields.url && checkedGroup.value === 'url') {
+    handleValidateFullUrl()
+  } else {
+    handleValidateCustomUrl()
+  }
+}
+
+const generateServiceName = (): string => {
```

Comment:
suggestion: add a comment explaining what you're doing here, possibly include sample generated name.